### PR TITLE
BREAKING: chore: remove downlevel-dts

### PR DESCRIPTION
**Pull request by yue4u**
BREAKING: TypeScript 3.9 is released in [May 2020](https://github.com/microsoft/TypeScript/releases/tag/v3.9.2). It does not make sense to support version before that.

1. Renamed all `.ts` -> `.d.ts` inside `types-*` packages
2. moved src -> types
3. remove building downlevel-dts but keeps files, ignores, etc

Work towards https://github.com/pixiv/three-vrm/pull/1293. This should also bring down build time a lot.

### CODE REVIEW

File: packages/three-vrm-core/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+types/
```
Comment (yue4u) [inline packages/three-vrm-core/.gitignore]: ~~Temp git ignore. After https://github.com/pixiv/three-vrm/pull/1293 types will end up in the lib dir~~ This is outdated

**Comment by 0b5vr** [comment]
@yue4u would you check `.eslintignore` and `.gitignore` ?
I searched the entire repo for `ts*` and there are these leftovers.

**Comment by yue4u** [comment]
> @yue4u would you check `.eslintignore` and `.gitignore` ? I searched the entire repo for `ts*` and there are these leftovers.

a1167dd9ed2f385f8d18dd56218a0c5ae576a71b

**Comment by 0b5vr** [comment]
LGTM! thank you!

**Review by 0b5vr — APPROVED** [review]

**Pull request merged**